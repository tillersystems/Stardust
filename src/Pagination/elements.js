import styled, { css } from 'styled-components';

export const Container = styled.div`
  height: 4rem;
  width: 100%;
  display: flex;
  margin-top: ${({ theme: { dimensions } }) => dimensions.medium};
  align-items: center;
  
  ${({ align }) =>
    align === 'left' &&
    css`
      justify-content: flex-start;
    `}
  ${({ align }) =>
    align === 'right' &&
    css`
      justify-content: flex-end;
    `}
  ${({ align }) =>
    align === 'center' &&
    css`
      justify-content: center;
    `}
`;

export const PaginationContainer = styled.div`
  width: auto;
  height: 4rem;
  display: flex;
  align-items: center;
  border-radius: ${({ theme: { dimensions } }) => dimensions.radius};
  border: 1px solid ${({ theme: { palette } }) => palette.lightGrey};
  background-color: ${({ theme: { palette } }) => palette.white};
`;

export const PaginationGroup = styled.div`
  height: 3rem;
  display: flex;
`;

export const PaginationItem = styled.div`
  border-radius: ${({ theme: { dimensions } }) => dimensions.radius};
  background-color: ${({ theme: { palette } }) => palette.white};
  height: 3rem;
  width: 3rem;
  line-heigh: 3rem;
  display: flex;
  justify-content: center;
  align-items: center;
  color: ${({ theme: { palette } }) => palette.mediumGrey};
  cursor: pointer;

  &:hover {
    background-color: ${({ theme: { palette } }) => palette.lightGrey};
  }

  ${({ active }) =>
    active &&
    css`
      background-color: ${({ theme: { palette } }) => palette.primary.default};
      color: ${({ theme: { palette } }) => palette.white};
      &:hover {
        background-color: ${({ theme: { palette } }) => palette.primary.darker};
      }
    `}
`;

export const BreakItem = styled.div`
  background-color: ${({ theme: { palette } }) => palette.white};
  height: 3rem;
  width: 3rem;
  display: flex;
  justify-content: center;
  align-items: center;
  color: ${({ theme: { palette } }) => palette.mediumGrey};
`;

export const ArrowContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 3rem;
  width: 3rem;
  ${({ side }) =>
    side === 'left' &&
    css`
      border-right: 1px solid ${({ theme: { palette } }) => palette.lightGrey};
      margin-right: 1rem;
    `}

  ${({ side }) =>
    side === 'right' &&
    css`
      border-left: 1px solid ${({ theme: { palette } }) => palette.lightGrey};
      margin-left: 1rem;
    `}

  cursor: pointer;
`;
