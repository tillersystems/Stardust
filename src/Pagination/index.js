import React, { useState } from 'react';
import PropTypes from 'prop-types';

import { Icon, Theme } from '..';

import {
  Container,
  PaginationContainer,
  PaginationItem,
  PaginationGroup,
  ArrowContainer,
  BreakItem,
} from './elements';

/**
 * Pagination
 *
 * This component is in charge of displaying
 * a Pagination
 *
 * @param {number} pageCount - total number of pages in the pagination.
 * @param {number} pageRangeDisplayed - displayed page range.
 * @param {number} marginPageDisplayed - page displayed on the pagination marges.
 * @param {string} breakLabel - default '...'. Label when you need a break for long pagination.
 * @param {func} getRequestedPageNumber - component callback function.
 * @param {string} align - component alignment. oneOf ['left', 'center', 'right'].
 *
 * @return {jsx}
 */
const Pagination = ({
  pageCount,
  pageRangeDisplayed,
  marginPageDisplayed,
  breakLabel,
  getRequestedPageNumber,
  align,
}) => {
  /**
   * Component initial state.
   */
  const [currentPage, setCurrentPage] = useState(1);

  /**
   * Page item click callback.
   *
   * @param {number} index
   */
  const handlePageItemClicked = index => {
    if (currentPage === index) return;

    setCurrentPage(index);
    getRequestedPageNumber(currentPage);
  };

  /**
   * Handle click on previous page.
   */
  const handlePrevious = () => {
    setCurrentPage(currentPage === 1 ? 1 : currentPage - 1);
    getRequestedPageNumber(currentPage);
  };

  /**
   * Handle click on next page.
   */
  const handleNextPage = () => {
    setCurrentPage(currentPage === pageCount ? pageCount : currentPage + 1);
    getRequestedPageNumber(currentPage);
  };

  /**
   * Return a pagination item.
   *
   * @param {number} index
   *
   * @return {jsx}
   */
  const getPaginationItem = index => {
    return (
      <PaginationItem
        key={index}
        active={currentPage === index}
        onClick={() => handlePageItemClicked(index)}
        data-testid={`page-${index}`}
      >
        {index}
      </PaginationItem>
    );
  };

  /**
   * Render the pagination.
   *
   * @return {Array}
   */
  const renderPagination = () => {
    const items = [];

    if (pageCount <= pageRangeDisplayed) {
      for (let i = 1; i <= pageCount; i++) {
        items.push(getPaginationItem(i));
      }
    } else {
      let left = pageRangeDisplayed / 2;
      let right = pageRangeDisplayed - left;
      let index;
      let paginationBreak;

      if (currentPage > pageCount - left) {
        right = pageCount - currentPage;
        left = pageRangeDisplayed - right;
      } else if (currentPage < left) {
        left = currentPage;
        right = pageRangeDisplayed - left;
      }

      for (index = 1; index <= pageCount; index++) {
        if (index <= marginPageDisplayed) {
          items.push(getPaginationItem(index));
          continue;
        }

        if (index > pageCount - marginPageDisplayed) {
          items.push(getPaginationItem(index));
          continue;
        }

        if (index >= currentPage - left && index <= currentPage + right) {
          items.push(getPaginationItem(index));
          continue;
        }

        if (breakLabel && items[items.length - 1] !== paginationBreak) {
          paginationBreak = <BreakItem key={index}>{breakLabel}</BreakItem>;
          items.push(paginationBreak);
        }
      }
    }

    return items;
  };

  /**
   * Render the component.
   *
   * @return {jsx}
   */
  return (
    <Container align={align} data-testid="pagination-main-container">
      <PaginationContainer data-testid="pagination-container">
        <ArrowContainer side="left" onClick={handlePrevious} data-testid="arrow-left">
          <Icon name="chevron-left" color={Theme.palette.lightGrey} />
        </ArrowContainer>
        <PaginationGroup>{renderPagination()}</PaginationGroup>
        <ArrowContainer side="right" onClick={handleNextPage} data-testid="arrow-right">
          <Icon name="chevron-right" color={Theme.palette.lightGrey} />
        </ArrowContainer>
      </PaginationContainer>
    </Container>
  );
};

/**
 * Pagination component props validation.
 */
const { number, string, func, oneOf } = PropTypes;

Pagination.propTypes = {
  pageCount: number,
  pageRangeDisplayed: number,
  marginPageDisplayed: number,
  breakLabel: string,
  align: oneOf(['left', 'right', 'center']),
  getRequestedPageNumber: func,
};

/**
 * Pagination default props.
 */
Pagination.defaultProps = {
  pageCount: 1,
  pageRangeDisplayed: 2,
  marginPageDisplayed: 1,
  breakLabel: '...',
  align: 'right',
  getRequestedPageNumber: () => {},
};

Pagination.displayName = 'Pagination';

export default Pagination;
