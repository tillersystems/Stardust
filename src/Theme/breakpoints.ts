/**
 * Defines application breakpoints.
 */

const xs = 0;
const sm = 576;
const md = 768;
const lg = 992;
const xl = 1200;

export const breakpoints = {
  xs,
  sm,
  md,
  lg,
  xl,
};
