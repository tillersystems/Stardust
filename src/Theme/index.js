/**
 * Theme exported to the provider
 * This is globally available in styled-components when interpolating a function like so:
 * ${(props) => props.theme}
 * Or using import { withTheme } from 'styled-components';
 */

import { breakpoints } from './breakpoints';
import { palette } from './palette';
import { dimensions } from './dimensions';
import { fonts } from './fonts';

const Theme = {
  breakpoints,
  palette,
  dimensions,
  fonts,
};

export default Theme;
