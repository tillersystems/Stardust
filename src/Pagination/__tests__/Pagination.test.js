import React from 'react';
import { fireEvent } from 'react-testing-library';

import Pagination from '..';
import Theme from '../../Theme';

/**
 * Test suit for Pagination component.
 */
describe('<Pagination />', () => {
  test('should render without a problem', () => {
    const { container } = render(<Pagination pageCount={10} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('should render without a problem without large number of pages', () => {
    const { container } = render(<Pagination pageCount={2} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('should render whithout problem on center', () => {
    const { container } = render(<Pagination pageCount={2} align="center" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('should render whithout problem on left', () => {
    const { container } = render(<Pagination pageCount={2} align="left" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('should select a page', () => {
    const { getByText } = render(<Pagination pageCount={10} />);

    const selectedPage = getByText('2');

    fireEvent.click(selectedPage);

    expect(selectedPage).toHaveStyleRule('background-color', Theme.palette.primary.default);
  });

  test('should select the last page', () => {
    const { getByText } = render(<Pagination pageCount={10} />);

    const selectedPage = getByText('10');

    fireEvent.click(selectedPage);

    expect(selectedPage).toHaveStyleRule('background-color', Theme.palette.primary.default);
  });

  test('should select a page in the middle', () => {
    const { getByText, getByTestId } = render(<Pagination pageCount={10} />);

    const arrowRight = getByTestId('arrow-right');

    fireEvent.click(arrowRight);
    fireEvent.click(arrowRight);
    fireEvent.click(arrowRight);
    fireEvent.click(arrowRight);

    const selectedPage = getByText('5');

    expect(selectedPage).toHaveStyleRule('background-color', Theme.palette.primary.default);
  });

  test('should go to next page on arrow right click', () => {
    const { getByTestId, getByText } = render(<Pagination pageCount={10} />);

    const arrowRight = getByTestId('arrow-right');
    const selectedPage = getByText('2');

    fireEvent.click(arrowRight);

    expect(selectedPage).toHaveStyleRule('background-color', Theme.palette.primary.default);
  });

  test('should go to previous page on arrow left click', () => {
    const { getByTestId, getByText } = render(<Pagination pageCount={10} />);

    const arrowRight = getByTestId('arrow-right');
    const arrowLeft = getByTestId('arrow-left');
    const firstPage = getByText('1');

    fireEvent.click(arrowRight);
    fireEvent.click(arrowLeft);

    expect(firstPage).toHaveStyleRule('background-color', Theme.palette.primary.default);
  });

  test('should call callback function', () => {
    const spy = jest.fn();

    const { getByTestId } = render(<Pagination pageCount={10} getRequestedPageNumber={spy} />);

    const arrowRight = getByTestId('arrow-right');

    fireEvent.click(arrowRight);

    expect(spy).toHaveBeenCalled();
  });
});
