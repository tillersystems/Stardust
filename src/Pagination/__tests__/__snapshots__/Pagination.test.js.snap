// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> should render whithout problem on center 1`] = `
.c3 {
  display: inline-block;
  border-image-source: none;
  border-image-width: 0;
  padding: 0;
  font-size: 20;
  line-height: 20;
}

.c0 {
  height: 4rem;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 1.5rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  width: auto;
  height: 4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.4rem;
  border: 1px solid hsl(207,22%,90%);
  background-color: hsl(0,100%,100%);
}

.c4 {
  height: 3rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 {
  border-radius: 0.4rem;
  background-color: hsl(0,100%,100%);
  height: 3rem;
  width: 3rem;
  line-heigh: 3rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: hsl(206,23%,69%);
  cursor: pointer;
  background-color: hsl(200,74%,46%);
  color: hsl(0,100%,100%);
}

.c5:hover {
  background-color: hsl(207,22%,90%);
}

.c5:hover {
  background-color: #14638a;
}

.c6 {
  border-radius: 0.4rem;
  background-color: hsl(0,100%,100%);
  height: 3rem;
  width: 3rem;
  line-heigh: 3rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: hsl(206,23%,69%);
  cursor: pointer;
}

.c6:hover {
  background-color: hsl(207,22%,90%);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 3rem;
  width: 3rem;
  border-right: 1px solid hsl(207,22%,90%);
  margin-right: 1rem;
  cursor: pointer;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 3rem;
  width: 3rem;
  border-left: 1px solid hsl(207,22%,90%);
  margin-left: 1rem;
  cursor: pointer;
}

<div
  class="c0"
  data-testid="pagination-main-container"
>
  <div
    class="c1"
    data-testid="pagination-container"
  >
    <div
      class="c2"
      data-testid="arrow-left"
    >
      <div
        class="c3"
        data-testid="iconContainer"
        height="20"
        width="20"
      >
        <svg
          data-testid="iconSvg"
          height="20"
          style="display: block;"
          viewBox="0 0 512 512"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M335.085 368.915c8.331 8.331 8.331 21.839 0 30.17-8.331 8.331-21.839 8.331-30.17 0l-128-128c-8.331-8.331-8.331-21.839 0-30.17l128-128c8.331-8.331 21.839-8.331 30.17 0 8.331 8.331 8.331 21.839 0 30.17L222.17 256l112.915 112.915z"
            data-testid="iconSvgPath"
            fill="hsl(207,22%,90%)"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
    <div
      class="c4"
    >
      <div
        class="c5"
        data-testid="page-1"
      >
        1
      </div>
      <div
        class="c6"
        data-testid="page-2"
      >
        2
      </div>
    </div>
    <div
      class="c7"
      data-testid="arrow-right"
    >
      <div
        class="c3"
        data-testid="iconContainer"
        height="20"
        width="20"
      >
        <svg
          data-testid="iconSvg"
          height="20"
          style="display: block;"
          viewBox="0 0 512 512"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M176.915 143.085c-8.331-8.331-8.331-21.839 0-30.17 8.331-8.331 21.839-8.331 30.17 0l128 128c8.331 8.331 8.331 21.839 0 30.17l-128 128c-8.331 8.331-21.839 8.331-30.17 0-8.331-8.331-8.331-21.839 0-30.17L289.83 256 176.915 143.085z"
            data-testid="iconSvgPath"
            fill="hsl(207,22%,90%)"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Pagination /> should render whithout problem on left 1`] = `
.c3 {
  display: inline-block;
  border-image-source: none;
  border-image-width: 0;
  padding: 0;
  font-size: 20;
  line-height: 20;
}

.c0 {
  height: 4rem;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 1.5rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c1 {
  width: auto;
  height: 4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.4rem;
  border: 1px solid hsl(207,22%,90%);
  background-color: hsl(0,100%,100%);
}

.c4 {
  height: 3rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 {
  border-radius: 0.4rem;
  background-color: hsl(0,100%,100%);
  height: 3rem;
  width: 3rem;
  line-heigh: 3rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: hsl(206,23%,69%);
  cursor: pointer;
  background-color: hsl(200,74%,46%);
  color: hsl(0,100%,100%);
}

.c5:hover {
  background-color: hsl(207,22%,90%);
}

.c5:hover {
  background-color: #14638a;
}

.c6 {
  border-radius: 0.4rem;
  background-color: hsl(0,100%,100%);
  height: 3rem;
  width: 3rem;
  line-heigh: 3rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: hsl(206,23%,69%);
  cursor: pointer;
}

.c6:hover {
  background-color: hsl(207,22%,90%);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 3rem;
  width: 3rem;
  border-right: 1px solid hsl(207,22%,90%);
  margin-right: 1rem;
  cursor: pointer;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 3rem;
  width: 3rem;
  border-left: 1px solid hsl(207,22%,90%);
  margin-left: 1rem;
  cursor: pointer;
}

<div
  class="c0"
  data-testid="pagination-main-container"
>
  <div
    class="c1"
    data-testid="pagination-container"
  >
    <div
      class="c2"
      data-testid="arrow-left"
    >
      <div
        class="c3"
        data-testid="iconContainer"
        height="20"
        width="20"
      >
        <svg
          data-testid="iconSvg"
          height="20"
          style="display: block;"
          viewBox="0 0 512 512"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M335.085 368.915c8.331 8.331 8.331 21.839 0 30.17-8.331 8.331-21.839 8.331-30.17 0l-128-128c-8.331-8.331-8.331-21.839 0-30.17l128-128c8.331-8.331 21.839-8.331 30.17 0 8.331 8.331 8.331 21.839 0 30.17L222.17 256l112.915 112.915z"
            data-testid="iconSvgPath"
            fill="hsl(207,22%,90%)"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
    <div
      class="c4"
    >
      <div
        class="c5"
        data-testid="page-1"
      >
        1
      </div>
      <div
        class="c6"
        data-testid="page-2"
      >
        2
      </div>
    </div>
    <div
      class="c7"
      data-testid="arrow-right"
    >
      <div
        class="c3"
        data-testid="iconContainer"
        height="20"
        width="20"
      >
        <svg
          data-testid="iconSvg"
          height="20"
          style="display: block;"
          viewBox="0 0 512 512"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M176.915 143.085c-8.331-8.331-8.331-21.839 0-30.17 8.331-8.331 21.839-8.331 30.17 0l128 128c8.331 8.331 8.331 21.839 0 30.17l-128 128c-8.331 8.331-21.839 8.331-30.17 0-8.331-8.331-8.331-21.839 0-30.17L289.83 256 176.915 143.085z"
            data-testid="iconSvgPath"
            fill="hsl(207,22%,90%)"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Pagination /> should render without a problem 1`] = `
.c3 {
  display: inline-block;
  border-image-source: none;
  border-image-width: 0;
  padding: 0;
  font-size: 20;
  line-height: 20;
}

.c0 {
  height: 4rem;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 1.5rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c1 {
  width: auto;
  height: 4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.4rem;
  border: 1px solid hsl(207,22%,90%);
  background-color: hsl(0,100%,100%);
}

.c4 {
  height: 3rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 {
  border-radius: 0.4rem;
  background-color: hsl(0,100%,100%);
  height: 3rem;
  width: 3rem;
  line-heigh: 3rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: hsl(206,23%,69%);
  cursor: pointer;
  background-color: hsl(200,74%,46%);
  color: hsl(0,100%,100%);
}

.c5:hover {
  background-color: hsl(207,22%,90%);
}

.c5:hover {
  background-color: #14638a;
}

.c6 {
  border-radius: 0.4rem;
  background-color: hsl(0,100%,100%);
  height: 3rem;
  width: 3rem;
  line-heigh: 3rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: hsl(206,23%,69%);
  cursor: pointer;
}

.c6:hover {
  background-color: hsl(207,22%,90%);
}

.c7 {
  background-color: hsl(0,100%,100%);
  height: 3rem;
  width: 3rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: hsl(206,23%,69%);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 3rem;
  width: 3rem;
  border-right: 1px solid hsl(207,22%,90%);
  margin-right: 1rem;
  cursor: pointer;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 3rem;
  width: 3rem;
  border-left: 1px solid hsl(207,22%,90%);
  margin-left: 1rem;
  cursor: pointer;
}

<div
  class="c0"
  data-testid="pagination-main-container"
>
  <div
    class="c1"
    data-testid="pagination-container"
  >
    <div
      class="c2"
      data-testid="arrow-left"
    >
      <div
        class="c3"
        data-testid="iconContainer"
        height="20"
        width="20"
      >
        <svg
          data-testid="iconSvg"
          height="20"
          style="display: block;"
          viewBox="0 0 512 512"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M335.085 368.915c8.331 8.331 8.331 21.839 0 30.17-8.331 8.331-21.839 8.331-30.17 0l-128-128c-8.331-8.331-8.331-21.839 0-30.17l128-128c8.331-8.331 21.839-8.331 30.17 0 8.331 8.331 8.331 21.839 0 30.17L222.17 256l112.915 112.915z"
            data-testid="iconSvgPath"
            fill="hsl(207,22%,90%)"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
    <div
      class="c4"
    >
      <div
        class="c5"
        data-testid="page-1"
      >
        1
      </div>
      <div
        class="c6"
        data-testid="page-2"
      >
        2
      </div>
      <div
        class="c7"
      >
        ...
      </div>
      <div
        class="c6"
        data-testid="page-10"
      >
        10
      </div>
    </div>
    <div
      class="c8"
      data-testid="arrow-right"
    >
      <div
        class="c3"
        data-testid="iconContainer"
        height="20"
        width="20"
      >
        <svg
          data-testid="iconSvg"
          height="20"
          style="display: block;"
          viewBox="0 0 512 512"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M176.915 143.085c-8.331-8.331-8.331-21.839 0-30.17 8.331-8.331 21.839-8.331 30.17 0l128 128c8.331 8.331 8.331 21.839 0 30.17l-128 128c-8.331 8.331-21.839 8.331-30.17 0-8.331-8.331-8.331-21.839 0-30.17L289.83 256 176.915 143.085z"
            data-testid="iconSvgPath"
            fill="hsl(207,22%,90%)"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Pagination /> should render without a problem without large number of pages 1`] = `
.c3 {
  display: inline-block;
  border-image-source: none;
  border-image-width: 0;
  padding: 0;
  font-size: 20;
  line-height: 20;
}

.c0 {
  height: 4rem;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 1.5rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c1 {
  width: auto;
  height: 4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.4rem;
  border: 1px solid hsl(207,22%,90%);
  background-color: hsl(0,100%,100%);
}

.c4 {
  height: 3rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 {
  border-radius: 0.4rem;
  background-color: hsl(0,100%,100%);
  height: 3rem;
  width: 3rem;
  line-heigh: 3rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: hsl(206,23%,69%);
  cursor: pointer;
  background-color: hsl(200,74%,46%);
  color: hsl(0,100%,100%);
}

.c5:hover {
  background-color: hsl(207,22%,90%);
}

.c5:hover {
  background-color: #14638a;
}

.c6 {
  border-radius: 0.4rem;
  background-color: hsl(0,100%,100%);
  height: 3rem;
  width: 3rem;
  line-heigh: 3rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: hsl(206,23%,69%);
  cursor: pointer;
}

.c6:hover {
  background-color: hsl(207,22%,90%);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 3rem;
  width: 3rem;
  border-right: 1px solid hsl(207,22%,90%);
  margin-right: 1rem;
  cursor: pointer;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 3rem;
  width: 3rem;
  border-left: 1px solid hsl(207,22%,90%);
  margin-left: 1rem;
  cursor: pointer;
}

<div
  class="c0"
  data-testid="pagination-main-container"
>
  <div
    class="c1"
    data-testid="pagination-container"
  >
    <div
      class="c2"
      data-testid="arrow-left"
    >
      <div
        class="c3"
        data-testid="iconContainer"
        height="20"
        width="20"
      >
        <svg
          data-testid="iconSvg"
          height="20"
          style="display: block;"
          viewBox="0 0 512 512"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M335.085 368.915c8.331 8.331 8.331 21.839 0 30.17-8.331 8.331-21.839 8.331-30.17 0l-128-128c-8.331-8.331-8.331-21.839 0-30.17l128-128c8.331-8.331 21.839-8.331 30.17 0 8.331 8.331 8.331 21.839 0 30.17L222.17 256l112.915 112.915z"
            data-testid="iconSvgPath"
            fill="hsl(207,22%,90%)"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
    <div
      class="c4"
    >
      <div
        class="c5"
        data-testid="page-1"
      >
        1
      </div>
      <div
        class="c6"
        data-testid="page-2"
      >
        2
      </div>
    </div>
    <div
      class="c7"
      data-testid="arrow-right"
    >
      <div
        class="c3"
        data-testid="iconContainer"
        height="20"
        width="20"
      >
        <svg
          data-testid="iconSvg"
          height="20"
          style="display: block;"
          viewBox="0 0 512 512"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M176.915 143.085c-8.331-8.331-8.331-21.839 0-30.17 8.331-8.331 21.839-8.331 30.17 0l128 128c8.331 8.331 8.331 21.839 0 30.17l-128 128c-8.331 8.331-21.839 8.331-30.17 0-8.331-8.331-8.331-21.839 0-30.17L289.83 256 176.915 143.085z"
            data-testid="iconSvgPath"
            fill="hsl(207,22%,90%)"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
